onnxruntime/test/providers/cuda/cuda_mempool_arena_test.cc

```diff
@@ -22,10 +22,17 @@ namespace test {
 
 // --------- Helpers ---------
 
+// cuda errors are sticky and may affect subsequent API calls.
+// we want to clear the error if when supported check fails.
+void ClearCudaError() {
+  ORT_IGNORE_RETURN_VALUE(::cudaGetLastError());
+}
+
 static bool IsCudaMemPoolSupported() {
   int ort_cuda_rt_version = 0;
   cudaError_t cuda_status = cudaRuntimeGetVersion(&ort_cuda_rt_version);
   if (cuda_status != cudaSuccess) {
+    ClearCudaError();
```

Comment:
Just curious - do we get cuda errors just by querying runtime and driver versions on some machines ?